Gems/PhysX/Code/Source/EditorPrismaticJointComponent.cpp

```diff
@@ -0,0 +1,217 @@
+        {
+            return m_config.m_forceMax;
+        }
+        else if (parameterName == PhysX::JointsComponentModeCommon::ParamaterNames::MaxTorque)
+        {
+            return m_config.m_torqueMax;
+        }
+        else if (parameterName == PhysX::JointsComponentModeCommon::ParamaterNames::Damping)
+        {
+            return m_linearLimit.m_standardLimitConfig.m_damping;
+        }
+        else if (parameterName == PhysX::JointsComponentModeCommon::ParamaterNames::Stiffness)
+        {
+            return m_linearLimit.m_standardLimitConfig.m_stiffness;
+        }
+
+        return 0.0f;
+    }
+
+    LinearLimitsFloatPair EditorPrismaticJointComponent::GetLinearValuePair(const AZStd::string& parameterName)
+    {
+        if (parameterName == PhysX::JointsComponentModeCommon::ParamaterNames::LinearLimits)
+        {
+            return LinearLimitsFloatPair(m_linearLimit.m_limitUpper, m_linearLimit.m_limitLower);
+        }
+
+        return LinearLimitsFloatPair();
+    }
+
+    AZStd::vector<JointsComponentModeCommon::SubModeParamaterState> EditorPrismaticJointComponent::GetSubComponentModesState()
+    {
+        return {};
+    }
+
+    void EditorPrismaticJointComponent::SetBoolValue([[maybe_unused]] const AZStd::string& parameterName, [[maybe_unused]] bool value)
+    {
+    }
+
+    void EditorPrismaticJointComponent::SetLinearValue(const AZStd::string& parameterName, float value)
+    {
+        if (parameterName == PhysX::JointsComponentModeCommon::ParamaterNames::MaxForce)
+        {
+            m_config.m_forceMax = value;
+        }
+        else if (parameterName == PhysX::JointsComponentModeCommon::ParamaterNames::MaxTorque)
+        {
+            m_config.m_torqueMax = value;
+        }
+        else if (parameterName == PhysX::JointsComponentModeCommon::ParamaterNames::Damping)
+        {
+            m_linearLimit.m_standardLimitConfig.m_damping = value;
+        }
+        else if (parameterName == PhysX::JointsComponentModeCommon::ParamaterNames::Stiffness)
+        {
+            m_linearLimit.m_standardLimitConfig.m_stiffness = value;
+        }
+    }
+
+    void EditorPrismaticJointComponent::SetLinearValuePair(const AZStd::string& parameterName, const LinearLimitsFloatPair& valuePair)
+    {
+        if (parameterName == PhysX::JointsComponentModeCommon::ParamaterNames::LinearLimits)
+        {
+            m_linearLimit.m_limitUpper = valuePair.first;
+            m_linearLimit.m_limitLower = valuePair.second;
+        }
+    }
+
+    void EditorPrismaticJointComponent::DisplayEntityViewport(
+        const AzFramework::ViewportInfo& viewportInfo, AzFramework::DebugDisplayRequests& debugDisplay)
+    {
+        EditorJointComponent::DisplayEntityViewport(viewportInfo, debugDisplay);
+
+        if (!m_config.ShowSetupDisplay() &&
+            !m_config.m_inComponentMode)
+        {
+            return;
+        }
+
+        const float size = 1.0f;
+        const float alpha = 0.6f;
```

Comment:
question: It might be worth moving these to AZ_CVARs to we could tweak them if need be and not leave them hard-coded?

Reply:
yeah that's a good idea, but again I'd suggest we do it for all the joints in a separate change

Reply:
sure thing :+1: